pkg/server/helpers_test.go

```diff
@@ -159,3 +160,74 @@ func TestGetSandbox(t *testing.T) {
 		assert.Equal(t, test.expected, sb)
 	}
 }
+
+func TestNormalizeImageRef(t *testing.T) {
+	for _, ref := range []string{
+		"busybox",        // has nothing
+		"busybox:latest", // only has tag
+		"busybox@sha256:e6693c20186f837fc393390135d8a598a96a833917917789d63766cab6c59582", // only has digest
+		"library/busybox",                  // only has path
+		"docker.io/busybox",                // only has hostname
+		"docker.io/library/busybox",        // has no tag
+		"docker.io/busybox:latest",         // has no path
+		"library/busybox:latest",           // has no hostname
+		"docker.io/library/busybox:latest", // full reference
+		"gcr.io/library/busybox",           // gcr reference
+		"gcr.io/library/busybox:latest@sha256:e6693c20186f837fc393390135d8a598a96a833917917789d63766cab6c59582", // both tag and digest
+	} {
+		t.Logf("TestCase %q", ref)
+		normalized, err := normalizeImageRef(ref)
+		assert.NoError(t, err)
```

Comment:
But you do own part of the normalization logic in `normalizeImageRef()`... The fact that containerd can accept it doesn't mean the reference was normarlized correctly.